plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -3355,4 +3378,42 @@ private record FirstChangeSnapshot(Snapshot snapshot)
 
     private record UnknownTableChange()
             implements TableChangeInfo {}
+
+    private static <K, V> V getIncrementally(
+            Map<K, AtomicReference<V>> cache,
+            K key,
+            Predicate<V> isSufficient,
+            Supplier<V> loader,
+            Function<V, V> incrementalLoader,
+            BinaryOperator<V> merger)
+    {
+        AtomicReference<V> valueHolder = cache.computeIfAbsent(key, _ -> new AtomicReference<>());
+        V oldValue = valueHolder.get();
+        if (oldValue != null && isSufficient.test(oldValue)) {
+            return oldValue;
+        }
+
+        V newValue = oldValue == null ? loader.get() : incrementalLoader.apply(oldValue);
+
+        verifyNotNull(newValue, "loader returned null for %s", key);
+
+        V merged = merger.apply(oldValue, newValue);
+        if (!valueHolder.compareAndSet(oldValue, merged)) {
```

Comment:
`mergeColumnStatistics` is called 2 times here.

Reply:
yes, that is intentional, we need to merge with a different value in case a parallel load happened.
this code along the same lines as `io.trino.plugin.hive.metastore.cache.CachingHiveMetastore#getIncrementally`.
also, merging is a cheap operation

Reply:
> in case a parallel load happened.

We're not AFAIK in such a situation in the planning.
Advocating here to drop  the accumulateAndGet part to keep the code simple.